spec/services/shopify_voider_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'spec_helper'
+
+require_relative '../../lib/active_merchant/billing/gateways/shopify.rb'
+
+describe ShopifyVoider do
+  let(:transaction_id) { '0xDEADBEEF' }
+  let(:order_id) { 'order_id' }
+  let(:transaction_amount) { 1 }
+  let(:transaction) { double(:transaction, amount: transaction_amount, id: transaction_id) }
+  let(:refunder_instance) { double('refunder_instance', perform: true) }
+
+  before do
+    allow(ShopifyAPI::Transaction).to receive(:find).and_return(transaction)
+  end
+
+  subject { described_class.new(transaction_id, order_id) }
+
+  context '.initialize' do
+    it 'successfully does it\'s thing' do
+      expect(subject).to be_truthy
+    end
+  end
+
+  context '.perform' do
+    before do
+      allow(ShopifyRefunder).to receive(:new).and_return(refunder_instance)
```

Comment:
What are your feelings about injecting dependencies like `ShopifyRefunder` and `ShopifyAPI::Transaction` as a matter of style?